### ISSUE TITLE
Fetching game mode

### ISSUE THREAD

**Issue by FYPetro** [OsuCelebrity/OsuCelebrity#35]
Send `/np` while you're spectating a user who is playing a map, you get

![Something like this](http://puu.sh/lCk4I.png)

So if you detect "is watching" and the ending mode "<Taiko>", you will know the mode. This means you have to spam #spectate.

**Comment by reesebiz** [comment]
#specator is a public chat. I don't think it's a wise idea to spam that.

**Comment by FYPetro** [comment]
Each time the spectatee starts playing a new map, the osu client will use /np on the spectatee so we can listen only.  no need for /np I believe.

**Comment by reesebiz** [comment]
This is only if they have it enabled in the options: http://puu.sh/lCnTC/e65a683d1b.jpg

**Comment by FYPetro** [comment]
if they enabled it, we show their rank in the mode they are playing. If they disabled it, we only show their !gamemode rank.

**Comment by reesebiz** [comment]
Ok that sounds good. As long as we're not the ones spamming chat.

**Comment by Tillerino** [comment]
I don't think we can join #spectate through IRC and the spectatee needs to have supporter to auto-send the /np. And we don't get it if we start spectating after the map has started. This is all in all non-feasible IMO.

**Comment by marcostudios** [comment]
Only osu! chat op can access #spectator from IRC

**Comment by Tillerino** [comment]
What kind of chat op and what kind of #spectator? Their own or the one
they're watching?

**Comment by marcostudios** [comment]
only GMT/QAT and Tournament Staff (since the Chat Moderator group is merged into GMT and QAT can Moderate the chat) can access the #spectator they're watching (https://osu.ppy.sh/forum/p/4695229)

**Comment by Tillerino** [comment]
That sounds like an awfully complex solution that *might* work, but won't
in many cases. I've already talked to peppy about getting the current game
mode and before we hit a dead end there, we don't need to look into other
solutions.

**Comment by omkelderman** [comment]
if only ppy would add the gamemode to the BanchoBot-STAT-response

**Comment by omkelderman** [comment]
could it be possible to catch game mode with 0d534b8c6cd98894c790ad777d90b4c18f32e2da?

oh waait, that only works for maps that are specifically made for a certain gamemode i guess, but if its a standard you probably still have no clue, guess its a little bit more than nothing?
